src/persist-client/src/read.rs

```diff
@@ -303,6 +423,29 @@ where
         // because Listen replays them at the original fidelity.
         (updates, frontier)
     }
+
+    /// Returns a `HollowBatch` enriched with this metadata appropriate for
+    /// listening-style behavior.
+    #[instrument(level = "debug", skip_all, fields(shard = %self.machine.shard_id()))]
+    pub async fn listen_hollow_batch(&mut self) -> ReaderEnrichedHollowBatch<T> {
```

Comment:
let's replace the old codepaths, the new way is strictly better! concretely: replace `Listen::next`'s implementation with this (but keep the name `next`) plz

Reply:
👌

Reply:
To keep this PR as small as possible, I'll do this in a follow up. I have the code locally, just don't want to gum this one up any more than necessary.

Reply:
Hmmm can you at least push that local branch somewhere I can look at it then? As-is, I have to manually eyeball the old impl vs the new one to review it (instead of git doing that for me 😄).

Concretely, stuff like the `read.downgrade_since(as_of_moveable.clone()).await;` comment I left in the last round would jump out if the old impl was replaced by the new one

Reply:
Pushed all of that code up; makes the diff much noisier the way I have it. lmk if it's too chaotic and I can reorganize the commits to make it more legible.